From 2dad4af427a55079e8e2978099511084ebd549a9 Mon Sep 17 00:00:00 2001
From: Alexys Jacob <ultrabug@gentoo.org>
Date: Thu, 11 May 2017 18:24:43 +0200
Subject: [PATCH] fix scylla-housekeeping version detection to work with newer
 setuptools

Newer setuptools parse_version() don't like dashed version strings,
so we should trim it to avoid false negative version_compare() checks.

Signed-off-by: Alexys Jacob <ultrabug@gentoo.org>
---
 scylla-housekeeping | 14 ++++++++++++--
 1 file changed, 12 insertions(+), 2 deletions(-)

diff --git a/scylla-housekeeping b/scylla-housekeeping
index 01e170a9..796fc25b 100755
--- a/scylla-housekeeping
+++ b/scylla-housekeeping
@@ -69,13 +69,23 @@ def create_uuid_file(fl):
     with open(args.uuid_file, 'w') as myfile:
         myfile.write(str(uuid.uuid1()) + "\n")
 
+def sanitize_version(version):
+    """
+    Newer setuptools don't like dashed version strings, trim it to avoid
+    false negative version_compare() checks.
+    """
+    if version and '-' in version:
+        return version.split('-', 1)[0]
+    else:
+        return version
+
 def check_version(ar):
     if config and (not config.has_option("housekeeping", "check-version") or not config.getboolean("housekeeping", "check-version")):
         return
     if ar.version and ar.version != '':
-        current_version = ar.version
+        current_version = sanitize_version(ar.version)
     else:
-        current_version = get_api('/storage_service/scylla_release_version')
+        current_version = sanitize_version(get_api('/storage_service/scylla_release_version'))
         if current_version == "":
             # API is down, nothing to do
             return
-- 
2.12.2

